sched/signal/sig_nanosleep.c

```diff
@@ -261,6 +262,12 @@ int clock_nanosleep(clockid_t clockid, int flags,
 {
   int ret;
 
+  if (clockid < CLOCK_REALTIME || clockid > CLOCK_BOOTTIME)
+    {
+      set_errno(EINVAL);
```

Comment:
Please change the line 296. Also `clock_nanosleep` should not set errno according to the latest POSIX spec.